proxy-node-gateway/src/main/java/uk/gov/ida/notification/resources/EidasAuthnRequestResource.java

```diff
@@ -85,11 +85,11 @@ private void setResponseDataInSession(HttpSession session, EidasSamlParserRespon
         session.setAttribute(SESSION_KEY_HUB_REQUEST_ID, vspResponse.getRequestId());
     }
 
-    private EidasSamlParserResponse parseEidasRequest(String encodedEidasAuthnRequest) {
-        return eidasSamlParserService.parse(new EidasSamlParserRequest(encodedEidasAuthnRequest));
+    private EidasSamlParserResponse parseEidasRequest(String encodedEidasAuthnRequest, String sessionId) {
+        return eidasSamlParserService.parse(new EidasSamlParserRequest(encodedEidasAuthnRequest), sessionId);
     }
 
-    private AuthnRequestResponse generateHubRequestWithVSP() { return vspProxy.generateAuthnRequest(); }
+    private AuthnRequestResponse generateHubRequestWithVSP(String sessionId) { return vspProxy.generateAuthnRequest(sessionId); }
```

Comment:
Naming convention note - can we rename this method to `generateHubRequestWithVsp`? We should use consistent camel case naming